### PR TITLE
chore!: deprecate methods in AppShellSettings that require UI

### PR DESCRIPTION
**Pull request by haijian-vaadin**
## Description

There are some methods on `AppShellSettings` that only work with legacy bootstrapping, marked those as deprecated, and throw unsupported exceptions to not confuse users.

Fixes https://github.com/vaadin/fusion/issues/60

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/AppShellSettings.java [hunk 1]
```diff
@@ -411,7 +411,12 @@ public void addMetaTag(Position position, String name, String content) {
...
      *
-     * @return the instance used for configuring the loading indicator
+     * @return An optional instance used for configuring the loading indicator
+     *         or an empty optional if UI is not available.
+     * 
+     * @Deprecated It only works when useDeprecatedV14Bootstrapping is enabled.
+     *             Use a {@link UIInitListener} instead when there are
+     *             server-side views.
      */
     public Optional<LoadingIndicatorConfiguration> getLoadingIndicatorConfiguration() {
```
Comment (knoobie) [inline flow-server/src/main/java/com/vaadin/flow/server/AppShellSettings.java:421]: To communicate the deprecation `@Deprecated` could be added.

Comment (haijian-vaadin) [reply flow-server/src/main/java/com/vaadin/flow/server/AppShellSettings.java:422]: right, thanks for the catch, only added to the comment, forgot to add it to the method as well, now fixed.

**Review by platosha — APPROVED** [review]

File: flow-server/src/main/java/com/vaadin/flow/server/AppShellSettings.java [hunk 2]
```diff
@@ -411,26 +411,44 @@ public void addMetaTag(Position position, String name, String content) {
...
     /**
      * Returns the object used for configuring the push channel.
      *
-     * @return the instance used for push channel configuration
+     * @return An optional instance used for push channel configuration or an
+     *         empty optional if UI is not available.
+     * 
+     * @deprecated It only works when useDeprecatedV14Bootstrapping is enabled.
+     *             Use a {@link UIInitListener} instead when there are
+     *             server-side views.
```
Comment (pleku) [inline flow-server/src/main/java/com/vaadin/flow/server/AppShellSettings.java:449]: Same as above

**Review by pleku — CHANGES_REQUESTED** [review]

File: flow-server/src/main/java/com/vaadin/flow/server/AppShellSettings.java [hunk 3]
```diff
@@ -411,26 +411,44 @@ public void addMetaTag(Position position, String name, String content) {
     /**
      * Returns the configuration object for loading indicator.
      *
-     * @return the instance used for configuring the loading indicator
+     * @return An optional instance used for configuring the loading indicator
+     *         or an empty optional if UI is not available.
+     * 
+     * @deprecated It only works when useDeprecatedV14Bootstrapping is enabled.
+     *             Use a {@link UIInitListener} instead when there are
+     *             server-side views.
```
[the remark this answers is not in the document]
Comment (haijian-vaadin) [reply flow-server/src/main/java/com/vaadin/flow/server/AppShellSettings.java:419]: Yes, good point, throwing an unsupported exception sounds better.

[the remark this answers is not in the document]
Comment (haijian-vaadin) [reply flow-server/src/main/java/com/vaadin/flow/server/AppShellSettings.java]: now updated the code to throw an unsupported exception, can you help to take another look?

**Review by pleku — COMMENTED** [review]
Would it be big effort to drop in unit tests that validate the behavior ? If it is just a few lines to an existing test, I would go for it

**Review by pleku — APPROVED** [review]

**Comment by pleku** [comment]
Needs a rebase from master due to the client dependency update breaking gwt-compilation

**Pull request merged**